exwm-manage.el

```diff
@@ -665,6 +665,20 @@ border-width: %d; sibling: #x%x; stack-mode: %d"
       (exwm--log "id=#x%x" window)
       (exwm-manage--unmanage-window window t))))
 
+(defun exwm-manage--on-MapNotify (data _synthetic)
+  (let ((obj (make-instance 'xcb:UnmapNotify)))
+    (xcb:unmarshal obj data)
+    (with-slots (window) obj
+      (exwm--log "id=#x%x" window)
+      ;; With this we ensure that a "window hierarchy change" happens after
+      ;; mapping the window, as some servers (XQuartz) do not generate it.
+      (xcb:+request exwm--connection
+          (make-instance 'xcb:ConfigureWindow
+                         :window window
+                         :value-mask xcb:ConfigWindow:StackMode
+                         :stack-mode xcb:StackMode:Above))
```

Comment:
We may receive MapNotify from other sources.  So perhaps we should test whether the X window is in `exwm--id-buffer-alist`?